libraries/AP_InertialSensor/AP_InertialSensor_MPU9250.cpp

```diff
@@ -421,8 +417,14 @@ bool AP_InertialSensor_MPU9250::_hardware_init(void)
     // Chip reset
     uint8_t tries;
     for (tries = 0; tries<5; tries++) {
+        // reset device
+        _register_write(MPUREG_PWR_MGMT_1, BIT_PWR_MGMT_1_DEVICE_RESET);
+
         hal.scheduler->delay(100);
 
+        // disable I2C as recommended by the datasheet
+        _register_write(MPUREG_USER_CTRL, BIT_USER_CTRL_I2C_IF_DIS);
```

Comment:
Yes, we can read the AK8963 just fine.  However apparently there's a race between the initialization of INS and the Compass.... we are going to take a look there, but it seems orthogonal to this one.